### PR TITLE
[feature](docs) add docs for SHOW-CATALOG-RECYCLE-BIN

### PR DESCRIPTION
**Pull request by nextdreamblue**
Signed-off-by: nextdreamblue <zxw520blue1@163.com>

# Proposed changes

relation with: https://github.com/apache/doris/issues/8421

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/en/docs/sql-manual/sql-reference/Show-Statements/SHOW-CATALOG-RECYCLE-BIN.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+### Description
+
+This statement is used to display the dropped meta informations that can be recovered
+
+语法：
+
+```sql
+SHOW CATALOG RECYCLE BIN [ WHERE NAME [ = "name" | LIKE "name_matcher"] ]
+```
```
Comment (morningman) [inline docs/en/docs/sql-manual/sql-reference/Show-Statements/SHOW-CATALOG-RECYCLE-BIN.md:41]: Please give a result example to explain the columns in result.

Comment (morningman) [reply docs/en/docs/sql-manual/sql-reference/Show-Statements/SHOW-CATALOG-RECYCLE-BIN.md:41]: And also need to update `docs/sidebars.json`

Comment (nextdreamblue) [reply docs/en/docs/sql-manual/sql-reference/Show-Statements/SHOW-CATALOG-RECYCLE-BIN.md:41]: done

**Review by morningman — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.78 seconds
 load time: 440 seconds
 storage size: 17179362937 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221111072047_clickbench_pr_44154.html

**Pull request merged**